### PR TITLE
DS-775 STANFORD-420 | Single CTA block

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Create Single CTA block component (with dark mode styles)
- Install FontAwesome PRO Regular and Solid packages which replaces the free ones.
- Update ci.yaml to reference fontawesome package token and uses node v18 instead of v16
- Update Container package component to allow for adding margin top and bottom
- Update LinkButton component to display different icons depending on CTA type; add email, size, and variant (gradient or default) props

# Review By (Date)
- Soon so we can all use the pro FontAwesome package

# Criticality
- 6

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to https://news.stanford.edu/developreport/yvonne-test-standard-page
2. Look at the 5 Single CTA block examples
3. Test at all breakpoints and using dark mode
Screenshot 2024-08-08 at 10 56 56 AM
Screenshot 2024-08-08 at 10 57 02 AM
4. Test all hocus states
5. Inspect the images and see that the alt text is rendered.
6. Look at code

# Associated Issues and/or People
- JIRA ticket(s) - STANFORD-420

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,22 +12,24 @@ https://docs.squiz.net/component-service/latest/getting-started/index.html#insta
...
 ```bash
   cd components
 ```
 
-Install dependencies
+4. Install dependencies
 
 ```bash
-  npm install
+  FONTAWESOME_PACKAGE_TOKEN=YOUR_FA_PACKAGE_TOKEN npm install
```
Comment (yvonnetangsu) [inline README.md:32]: Updated the readme here

Comment (iamrentman) [reply README.md:32]: tysm!

**Review by iamrentman — APPROVED** [review]
GTG! 🚀

**Review by iamrentman — CHANGES_REQUESTED** [review]
When originally approving I hadn't looked at the editorial side of Matrix. I think we can make the CTA fields conditional based on the selection in your `ctaType` field so that all of the different URL fields aren't always displayed but the appropriate one is based on that selection. It'd just make the editorial UI cleaner. We can discuss.

**Comment by yvonnetangsu** [comment]
> When originally approving I hadn't looked at the editorial side of Matrix. I think we can make the CTA fields conditional based on the selection in your `ctaType` field so that all of the different URL fields aren't always displayed but the appropriate one is based on that selection. It'd just make the editorial UI cleaner. We can discuss.

Hi @iamrentman! This is ready for your review again. I have learned that using https://rjsf-team.github.io/react-jsonschema-form/ doesn't always work. Sometimes it works, and sometimes it doesn't even if I click the "Blank" example first before pasting. However, I have deployed this to the development environment and I believe it is working as intended. Here's my logic:

- If CTA type is "email", display the email address field
- For all other types (link and download), display the internal link, manual link, and new window checkbox fields

I also updated the descriptions in the manifest a bit so it's clearer. Let me know if this works for you.

**Review by iamrentman — APPROVED** [review]
Perf! Thanks for the conditional field update. 🚀

**Comment by yvonnetangsu** [comment]
Thanks, @iamrentman! I'm going to deploy this to `master`.

**Pull request merged**